lib/internal/util/iterable-weak-map.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const {
+  Set,
+  SetPrototypeAdd,
+  SetPrototypeDelete,
+  Symbol,
+  WeakMap,
+  WeakMapPrototypeGet,
+  WeakMapPrototypeDelete,
+  WeakMapPrototypeSet,
+} = primordials;
+
+// This class is modified from the example code in the WeakRefs specification:
+// https://github.com/tc39/proposal-weakrefs
+// Licensed under ECMA's MIT-style license, see:
+// https://github.com/tc39/ecma262/blob/master/LICENSE.md
+class IterableWeakMap {
+  #weakMap = new WeakMap();
+  #refSet = new Set();
+  #finalizationGroup = new global.FinalizationRegistry(cleanup);
+
+  set(key, value) {
```

Comment:
I don't understand how this works - this appears to create a new `WeakRef` each time the key is set even if the key "value" already exists in the map? Is this intentional?

Reply:
```
var m = new Map();
var o = {};
var r1 = new WeakRef(o);
var r2 = new WeakRef(o);
m.set(r1, "foo");
console.log(m.get(r1)); // foo 
console.log(m.get(r2)); // undefined
```

Reply:
> this appears to create a new WeakRef each time the key is set even if the key "value" already exists in the map? Is this intentional

I'll add unit tests for the `IterableWeakMap` implementation, and will make sure this case is covered.

Reply:
I'm saying - I don't understand how it works and to my understanding it's not supposed to. A test is fine, but a "this works because X" would also be fine :]

Reply:
@benjamingr I'm pretty sure you've identified a bug in the `IterableWeakMap` shared on the TC39 spec page -- we should only create a new `WeakRef` if there's not already an entry, otherwise we should perform an update.

Once I've fixed this, I'll share the updated code, and can provide a walk through of the logic if it's still confusing.

Reply:
Sure, feel free to ping me when updated and I'll take a look :]